### ISSUE TITLE
Unreliable, multiple acceptance tests when async is on

### ISSUE THREAD

**Issue by lukerollans** [keathley/wallaby#85]
Hey there!

I've been experiencing this for a while, but given it isn't important I haven't opened an issue. I consistently get timeout issues (and sometimes other, seemingly unrelated error messages) when I run my acceptance tests asynchronously using the `async` option.

If I turn off the async option everything is green but my suite takes about an extra 3-4 seconds to run at its current size. Currently sitting at about 200 tests all up, so it's not huge, but will continue to grow pretty steadily so I'm hoping to fix this issue now rather than later.

### Important: This only happens on CI.. which is an ec2 t2.small instance running in a Docker Compose container. Might a simply need to beef that up a bit?

Here's an example of two acceptance tests that reproduce the issue when run in isolation together.

### Test that a user can log in
```elixir
defmodule Sightseer.UserLogsInTest do
  use Sightseer.AcceptanceCase, async: true

  test "user logs in", %{session: session} do
    user = insert(:user, encrypted_password: Comeonin.Bcrypt.hashpwsalt("test"))

    session
    |> visit("/sessions/new")
    |> fill_in("Email address", with: user.email)
    |> fill_in("Password", with: "test")
    |> click_button("Submit")

    notice = session
    |> find(".alert-info")
    |> text

    assert notice == "Signed in successfully"
  end
end
```

### Test that a user can register
```elixir
defmodule Sightseer.UserSignsUpTest do
  use Sightseer.AcceptanceCase, async: true

  test "user signs up", %{session: session} do
    session
    |> visit("/signup")
    |> fill_in("First name", with: "Luke")
    |> fill_in("Email address", with: "some@email.com")
    |> fill_in("Password", with: "1234567madpasswordbro")
    |> click_button("Submit")

    notice = session
    |> find(".alert-info")
    |> text

    assert notice == "Successfully signed up!"
  end
end
```

### Error thrown on CI in the environment described above, when `async` is on
```
Generated sightseer app
 
  1) test user logs in up (Sightseer.UserLogsInTest)
     test/acceptance/user_logs_in_test.exs:4
     ** (exit) exited in: :gen_server.call(Wallaby.ServerPool, {:checkout, #Reference<0.0.1.94027>, true}, 5000)
         ** (EXIT) time out
     stacktrace:
       (stdlib) gen_server.erl:212: :gen_server.call/3
       (poolboy) src/poolboy.erl:55: :poolboy.checkout/3
       (wallaby) lib/wallaby.ex:34: Wallaby.start_session/1
       (sightseer) test/support/acceptance_case.ex:21: Sightseer.AcceptanceCase.__ex_unit_setup_0/1
       (sightseer) test/support/acceptance_case.ex:1: Sightseer.AcceptanceCase.__ex_unit__/2
       test/acceptance/user_logs_in_test.exs:1: Sightseer.UserLogsInTest.__ex_unit__/2
 
...................................................................................................................................................................................
 
Finished in 5.9 seconds
180 tests, 1 failure
```

Any insight in to how I might be able improve reliability here would be very much appreciated :)

**Comment by keathley** [comment]
Thanks for the super detailed issue @lukerollans :+1:

So the issue here is that pool boy, which is what wallaby uses to manage phantom workers, is running out of workers. When that happens the test that is trying to check out a worker blocks until it eventually times out and fails. Luckily, @stevegraham just added a fix for this here: https://github.com/keathley/wallaby/pull/84. I just released version 0.11. Would you mind pulling that down and seeing if you still have the same issues?

**Comment by lukerollans** [comment]
Yep, that did the trick
